src/webgpu/api/operation/render_pipeline/primitive_topology.spec.ts

```diff
@@ -0,0 +1,376 @@
+export const description = `Test primitive topology rendering.
+
+Draw a primitive using 6 vertices with each topology and check if the pixel is covered.
+
+Vertex sequence and coordinates are the same for each topology:
+  - Vertex buffer = [v1, v2, v3, v4, v5, v6]
+  - Topology = [point-list, line-list, line-strip, triangle-list, triangle-strip]
+
+Test locations are framebuffer coordinates:
+  - Pixel value { valid: green, invalid: black, format: 'rgba8unorm'}
+  - Test point is valid if the pixel value equals the covered pixel value at the test location.
+
+  Topology: point-list    Test points(s)
+
+  v2      v3      v6      Every vertex.
+
+
+
+  v1      v4      v5
+
+  Topology: line-list (3 lines)
+
+  v2       v4      v6
+  |        |       |      Center of each vertical line segment.
+  |        |       |
+  |        |       |
+  v1       v3      v5
+
+  Topology: line-strip (5 lines)
+
+  v2 ----- v3      v6
```

Comment:
Maybe we should test that the primitive restart correctly breaks the strip in two. Ditto for triangles.

Reply:
Done.